httpclient5/src/main/java/org/apache/hc/client5/http/HttpRequestRetryStrategy.java

```diff
@@ -75,6 +75,22 @@
      */
     boolean retryRequest(HttpResponse response, int execCount, HttpContext context);
 
+
+    /**
+     * Determines the retry interval between subsequent retries.
+     *
+     * @param request the request failed due to an I/O exception
+     * @param exception the exception that occurred
+     * @param execCount the number of times this method has been
+     *                  unsuccessfully executed
+     * @param context the context for the request execution
+     *
+     * @return the retry interval between subsequent retries
+     */
+    default TimeValue getRetryInterval(HttpRequest request, IOException exception, int execCount, HttpContext context) {
```

Comment:
@ajbaldocchi Please the existing `getRetryInterval` method for default implementation of this method.

Reply:
Could you please clarify the change you would like here?

Reply:
@ajbaldocchi My aplogies. Please disregard my comment. I overlooked that the new method takes `HttpRequest` as a parameter.